packages/openai/src/responses/openai-responses-language-model.test.ts

```diff
@@ -4957,6 +5026,43 @@ describe('OpenAIResponsesLanguageModel', () => {
         });
       });
     });
+
+    describe('apply_patch tool streaming', () => {
+      it('should handle apply_patch tool calls in streaming mode', async () => {
+        prepareChunksFixtureResponse('openai-apply-patch-tool.1');
+
+        const { stream } = await createModel('gpt-5.1-2025-11-13').doStream({
+          prompt: TEST_PROMPT,
+          tools: [
+            {
+              type: 'provider',
+              id: 'openai.apply_patch',
+              name: 'apply_patch',
+              args: {},
+            },
+          ],
+        });
+
+        const parts = await convertReadableStreamToArray(stream);
+        const toolCallParts = parts.filter(
+          part => part.type === 'tool-call' && part.toolName === 'apply_patch',
+        );
+
+        expect(toolCallParts.length).toBeGreaterThan(0);
+        const toolCall = toolCallParts[0];
+        if (toolCall && toolCall.type === 'tool-call') {
+          const input = JSON.parse(toolCall.input);
+          expect(input).toMatchObject({
+            callId: 'call_kA46f91ZwocQyMCKyyZqRyC5',
+            operation: {
+              type: 'create_file',
+              path: 'shopping-checklist.md',
+              diff: expect.stringContaining('Shopping Checklist'),
+            },
+          });
+        }
+      });
```

Comment:
logic like this is not great in tests in my experience (also above). usually dumb snapshots are easier to debug and to review and faster to write

might be good to make a smol follow up pr